### ISSUE TITLE
Issues about COCO eval and VOC2007 eval

### ISSUE THREAD

**Issue by lst627** [m1k2zoo/negbench#2]
Thank you for your great work and documentation! I have two questions regarding the evaluation on NegBench.

1. Following your instruction, now I can reproduce the result for openai-ViT-B/32 on COCO-MCQ and the total accuracy is 20.74%. However, the recall@1 and recall@5 on COCO are pretty low as follows:

{"coco-image_retrieval_recall@1": 0.0002398656797595322, "coco-image_retrieval_recall@5": 0.0010793955298140645, 
"coco-negated-image_retrieval_recall@1": 0.00019988806161563843, "coco-negated-image_retrieval_recall@5": 0.0008395298500545323}

2. For VOC2007 evaluation, I could not find the images listed in your csv in the downloaded VOC2007 dataset. Could you check whether the link in your document is correct?

**Comment by m1k2zoo** [comment]
Hi, thank you so much for your interest in our work! 

1. Let me cross-reference the instructions I uploaded with the local scripts I have and get back to you. If you get a chance, can you please tell me what Python command generated the COCO results? I will spend some time to inspect what might be going wrong tomorrow.  
2. You are right, thanks a lot for catching that! We used the VOC2007 test split in our NegBench evaluation, but I had linked to the trainval data by mistake. I have updated datasets.md to include the correct link. 


Here it is for your reference:
http://host.robots.ox.ac.uk/pascal/VOC/voc2007/VOCtest_06-Nov-2007.tar

**Comment by m1k2zoo** [comment]
I just cloned this repo again and ran the following evaluation script (ran it from the `/benchmarks` folder):

```
#!/bin/bash

# Set the base directory for data and logs. Users should update this to their directory structure.
IMAGE_METADATA_DIR="PATH_TO_IMAGE_METADATA"
LOGS_DIR="../logs"

# Model and pretrained options
MODEL="ViT-B-32"
MODEL_NAME="ViT_B_32_openai"
PRETRAINED_MODEL="openai"

# Dataset paths for images
COCO_MCQ="$IMAGE_METADATA_DIR/COCO_val_mcq_llama3.1_rephrased.csv"
VOC_MCQ="$IMAGE_METADATA_DIR/VOC2007_mcq_llama3.1_rephrased.csv"
COCO_RETRIEVAL="$IMAGE_METADATA_DIR/COCO_val_retrieval.csv"
COCO_NEGATED_RETRIEVAL="$IMAGE_METADATA_DIR/COCO_val_negated_retrieval_llama3.1_rephrased_affneg_true.csv"

# Activate the appropriate environment
conda activate clip_negation

# Set system limits
ulimit -S -n 100000

# Logs directory for this evaluation run
RUN_LOGS_DIR="$LOGS_DIR/evaluation"
mkdir -p "$RUN_LOGS_DIR"

# Image Evaluation
echo "Starting Image Evaluation..."
CUDA_VISIBLE_DEVICES=0 python -m src.evaluation.eval_negation \
    --model $MODEL \
    --pretrained $PRETRAINED_MODEL \
    --name "image_$MODEL_NAME" \
    --logs=$RUN_LOGS_DIR \
    --dataset-type csv \
    --csv-separator=, \
    --csv-img-key filepath \
    --csv-caption-key caption \
    --zeroshot-frequency 1 \
    --imagenet-val="$DATA_DIR/images/imagenet" \
    --coco-mcq=$COCO_MCQ \
    --voc2007-mcq=$VOC_MCQ \
    --coco-retrieval=$COCO_RETRIEVAL \
    --coco-negated-retrieval=$COCO_NEGATED_RETRIEVAL \
    --batch-size=64 \
    --workers=8
```

It seems to work on my end. Can you check your PyTorch version? This is mine:
```
python -c "import torch; print(torch.__version__)"
2.0.1+cu117

```
If you get a chance to download and run VOC2007, please let me know what MCQ and Retrieval numbers you get. Thanks!

**Comment by lst627** [comment]
Thank you so much for your quick response! 

I found that I did not set the PRETRAINED_MODEL by mistake! Now for openai-ViT-B-32, the COCO MCQ total accuracy is 16.27% (interestingly, this is lower than the previous result), and the COCO retrieval is back to normal as follows:
{"coco-image_retrieval_recall@1": 0.30355000495910645, "coco-image_retrieval_recall@5": 0.5476933121681213, "coco-negated-image_retrieval_recall@1": 0.2497001737356186, "coco-negated-image_retrieval_recall@5": 0.4792915880680084}

Additionally, I have downloaded and run your evaluation on VOC2007. The VOC2007 MCQ total accuracy is 14.47%, and it does not have a retrieval task.

**Comment by m1k2zoo** [comment]
Awesome, yes this perfectly matches my numbers now! You can see from Figure 4 in the paper that CLIP (which is openai-ViT-B-32) achieves around 16% on COCO. 14.47% on VOC2007 also matches what I have.

**Comment by lst627** [comment]
Thank you so much!
